examples/avro-cli.py

```diff
@@ -152,21 +176,21 @@ def consume(topic, conf):
 
 
 def main(args):
-    # handle common configs
-    conf = {'bootstrap.servers': args.bootstrap_servers,
-            'schema.registry.url': args.schema_registry}
-
+    registry_client = SchemaRegistryClient(url=args.schema_registry)
     if args.userinfo:
-        conf['schema.registry.basic.auth.credentials.source'] = 'USER_INFO'
-        conf['schema.registry.basic.auth.user.info'] = args.userinfo
+        registry_client.user_auth(args.userinfo.split(':'))
+
+    avro_serializer = AvroSerializer(registry_client, record_schema,
+                                     converter=UserConverter)
```

Comment:
by passing the class, rather than an instance of a class, you're relying on duck typing here, not that `UserConverter` implements `Converter`, which it doesn't.

Reply:
Ditto(method resolution hack)